### ISSUE TITLE
Problems accessing shares in Windows

### ISSUE THREAD

**Issue by Tiwas** [SickChill/SickChill#7004]
I'm running a windows server (it's not based on the server software, but on Windows 10 x64). On startup, I have some persistent shares - a few from a NAS (SMB) and a couple using sshfs. When the computer starts these are usually accessible, but not until the first time they're opened, and this is where the problems start.

Is it possible to create mount points from within sickchill or have it open mount points that are set up, but not active?

This is the current status:
![image](https://user-images.githubusercontent.com/17053402/106847471-3e920900-66af-11eb-8485-bbedaee5ea71.png)

The ones marked as "unavailable" will only need to be accessed once to change to "OK". If this is not possible using sickchill, are there any other known workarounds?

**Comment by quickdry** [comment]
could you delay sickchill service startup, run a script to access those shares and have them ready for when the service starts?

**Comment by Tiwas** [comment]
> could you delay sickchill service startup, run a script to access those shares and have them ready for when the service starts?

Thanks. I have a script already, and SC doesn't seem to mind if the shares are up before or after it starts. However, there are issues using a script for the sshfs shares in particular, and I suspect it has something to do with the vpn also being delayed (it takes time to connect properly). 

This is how it looks to me:
* Sickchill starts
* Mounts are accessible (doesn't matter if SC starts before or after)
* Random amount of seconds (seems to be less than 30), vpn goes from a waiting state to initiate
* External mounts become inaccessible, probably due to changing external IP
* Some time later, probably based on how many slots are left on the vpn and such, the vpn is connected and open
* Now, I can re-establish the connection to external shares

I don't think there's any way for me to do a post process on the shares, and even then I don't know how I should handle other events that might make them inaccessible (vpn going down, temp connection drop-out, whatever). My hunch, though, is that if SC would try to initiate a connection to a mount even if it's in an unavailable state (SC settings state which share to connect to) it would solve these problems automatically.

**Comment by miigotu** [comment]
@Tiwas what does your script look like that you use to ping the shares?

**Comment by miigotu** [comment]
And what command did you use to get that pretty output on share status?

**Comment by Tiwas** [comment]
@miigotu The status is done with a simple 'net use'. Or the following to use a script to reconnect them:
`for /f "TOKENS=1,2" %a in ('net use') do if "%a"=="Unavailable" echo %b`

This can run on a schedule. I'm sure there are excellent things to be done in linux, but if there are better ways in windows I haven't been able to find them.

**Comment by miigotu** [comment]
I didn't forget about you, just haven't come up with anything yet.

**Comment by Tiwas** [comment]
> I didn't forget about you, just haven't come up with anything yet.

Not worried about that, bud - and you *do* have more important things going on. Even babies rank higher than my shares occasionally becoming inaccessible 😲 I know, I know. Even I - a dog person - knows that 🤣

No, seriously. (I'm *not* saying my share take precedense over children) Take your time. I know it'll happen when you have time, and I have a semi-working pain-in-the-but solution (requires me to log in and manually start a job, so nothing will work for long if I don't - but that will remind me to log in :D ), so put it in the "nice to have [for some]" feature pile.

Also, in the example I gave above, the "echo %b" is just to echo the drive/mount not working. To *access* it, remove the echo.

Come to think of it - would it be feasible to add a pre-script? This page (yeah, I know - everyone but me probably knows how to do that already, just including it because *I* think I was pretty genius to think of it *before* 7am) mentions running cmd files on Windows: https://janakiev.com/blog/python-shell-commands/ . How about setting up one (or more) batch/console/whatever scripts to run before it does anything? Leaving it blank wouldn't do anything to system performance, adding it might ruin it (depending on what people put in it). Could be a pre-post-process script?

**Comment by miigotu** [comment]
We got a puppy too haha
![image](https://user-images.githubusercontent.com/390379/108023995-b3d7d500-6ff1-11eb-895f-af5d36e8aebb.png)

**Comment by Tiwas** [comment]
Ah, that's cute - and besides, all children should grow up with animals (read: dogs). It teaches them about empathy and caring, and responsibility. And they'll have a friend for life.

What breed is it? Kind of looks a little like a labrador, but also has some features looking like a bull-something? It's hard to tell which breed a puppy is. It's like for me to try to guess who's the parents of a newborn. The kid in the picture's a little easier to tell. Unless the postman looks a lot like you 🤣

**Comment by miigotu** [comment]
She's a mix of american staffordshire terrier, american pit bull, and a touch of boxer.

**Comment by Tiwas** [comment]
I might have found the solution to this, and have made the following reg-file
`;
; Please see the following thread for information
; https://github.com/billziss-gh/sshfs-win/issues/108
;
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\WinFsp\Services\sshfs]
"Recovery"=dword:00000001

`
